pkgs/tools/filesystems/bees/bees-service-wrapper

```diff
@@ -0,0 +1,215 @@
+#!@bash@/bin/bash
+PATH=@bash@/bin:@coreutils@/bin:@utillinux@/bin:@btrfsProgs@/bin:$PATH
+beesd_bin=@bees@/lib/bees/bees
+# PLEASE KEEP NIX-ISMS ABOVE THIS LINE TO EASE UPSTREAM MERGE
+#!/usr/bin/env bash
+
+shopt -s extglob
+
+# Upstream wrapper requires UUID to be used for configuration.
+
+# However, when declaratively describing a host, we may not know its UUID, and
+# shouldn't need to persist something that will differ between hosts built from
+# the same configuration template.
+
+# Thus, for using bees from NixOS, we have our own wrapper, which supports not
+# just UUID but any specification permitted by findmnt
+
+[[ $bees_debug ]] && { PS4=':${BASH_SOURCE##*/}:$LINENO+'; set -x; }
+
+usage() {
+  cat >&2 <<EOF
+Usage: ${BASH_SOURCE##*/} run|cleanup config-name|fsSpec idxSizeMB=...] [verbosity=...] [workDir=...] [-- daemon-options...]
+
+  fsSpec should be in a format recognized by findmnt. Alternately,
+  "config-name" may refer to a file that exists in ${bees_config_dir:-/etc/bees}
+  with a .conf extension; if that file does not specify UUID, findmnt will be
+  used in addition.
```

Comment:
Do you want to mention that the config will be sourced now, but it might move to key=value in the future?

Reply:
Hmm. Assuming this wrapper doesn't stay part of NixOS, plausibility of that future change is really up to bees upstream (discussion of upstreaming this is ongoing in Zygo/bees#80). I'm not sure I want to be making roadmap assertions about their project.

Reply:
«might break» is quite a weak statement, though

Reply:
How's this?

>  Note that while config files may presently use shell arithmetic, use of this
> functionality is not encouraged going forward: Setting `idxSizeMB=4096` is
> preferred over `DB_SIZE=$((1024*1024*1024*4))` or `DB_SIZE=$(( AL16M * 256 ))`,
> although both of these are presently supported.

Reply:
I like that wording